examples/introExample/src/main/java/io/realm/examples/intro/IntroExampleActivity.java

```diff
@@ -182,14 +186,20 @@ private String complexReadWrite() {
             // The field tempReference is annotated with @Ignore
             // Though we initially set its value to 42, it has
             // not been saved as part of the Person RealmObject:
-            assert(pers.getTempReference() == 0);
+            if (BuildConfig.DEBUG && pers.getTempReference() != 0) {
+                throw new AssertionError();
+            }
         }
 
         // Sorting
         RealmResults<Person> sortedPersons = realm.allObjects(Person.class);
         sortedPersons.sort("age", Sort.DESCENDING);
-        assert(realm.allObjects(Person.class).last().getName() == sortedPersons.first().getName());
-        status += "\nSorting " + sortedPersons.last().getName() + " == " + realm.allObjects(Person.class).first().getName();
+        if (BuildConfig.DEBUG &&
```

Comment:
I would probably remove these asserts all together from an intro example. They don't provide any value for how Realm works IMO?